### ISSUE TITLE
Optional chaining rule stopped working

### ISSUE THREAD

**Issue by EvanMorrison** [babel/eslint-plugin-babel#197]
Something has gotten out of sync between `eslint-plugin-babel`, `@babel/parser`, and/or `@babel/types` that has suddenly caused the `babel/no-unused-expressions` rule to incorrectly fail for optional chaining expressions. I found if I change the applicable function in [eslint-plugin-babel/rules/no-unused-expressions.js](https://github.com/babel/eslint-plugin-babel/blob/master/rules/no-unused-expressions.js) as follows, the linting works again:

```javascript
function isOptionalCallExpression(node) {
  return (
    !!node &&
    node.type === 'ExpressionStatement' &&
    node.expression.type === 'OptionalCallExpression'
  );
}
```
to 

```javascript
function isOptionalCallExpression(node) {
  return (
    !!node &&
    node.type === 'ExpressionStatement' &&
    node.expression.type === 'ChainExpression'
  );
}
```
I'm not experienced enough to know if that's the actual fix, or if it would cause other problems, but maybe it's a clue to what happened. I see that two dependencies of `babel-eslint` (`@babel/parser` and `@babel/types`) had minor releases 4 days ago. And at least some of the changes were related to code involving `optionalCallExpression` vs `chainExpression`.

For reference:

Examples of code that was passing last week, but which now incorrectly fails the no-unused-expressions rule:
```javascript
    this.systemNameSelect?.removeEventListener('mouseleave', this.listener);
    this.productsSelect?.removeEventListener('mouseleave', this.listener);
...
    const input = document.querySelector("form[name='advanced-search-form']").clientName;
    input?.focus();
```

I have the following relevant dev-dependencies in `package.json`
```json
    "@babel/cli": "7.8.4",
    "@babel/core": "7.9.0",
    "@babel/plugin-proposal-class-properties": "7.8.3",
    "@babel/plugin-transform-runtime": "7.9.0",
    "@babel/preset-env": "7.9.0",
    "@babel/preset-react": "7.9.4",
    "@babel/register": "7.9.0",
    "babel-eslint": "10.0.3",
    "babel-loader": "8.0.6",
    "eslint": "6.8.0",
    "eslint-config-prettier": "6.11.0",
    "eslint-config-standard": "14.1.1",
    "eslint-loader": "3.0.3",
    "eslint-plugin-babel": "5.3.0",
    "eslint-plugin-emotion": "10.0.0",
    "eslint-plugin-import": "2.20.0",
    "eslint-plugin-node": "11.0.0",
    "eslint-plugin-prettier": "3.1.3",
    "eslint-plugin-promise": "4.2.1",
    "eslint-plugin-react": "7.18.0",
    "eslint-plugin-react-hooks": "2.3.0",
    "eslint-plugin-standard": "4.0.1",
    "prettier": "2.0.5",
```

and the following is the relevant portion of `.eslintrc.js`
```javascript
  parser: 'babel-eslint',
  plugins: ['babel', 'emotion', 'import', 'node', 'promise', 'react', 'react-hooks', 'standard'],
  rules: {
    'no-unused-expressions': 0,
    'babel/no-unused-expressions': ['error', { allowTernary: true }],
...
```

But as I say, our configuration was working fine last week and we did not make any changes to it. For now we have turned off the rule, but obviously that is not a permanent solution.

Any guidance or ideas about how to proceed with this would be appreciated.

**Comment by JLHwung** [comment]
Duplicate of #196.
